### ISSUE TITLE
[Feature] Apply style to search bar in issuer list

### ISSUE THREAD

**Issue by raginmari** [Adyen/adyen-ios#1174]
**Is your feature request related to a problem? Please describe.**
When styling various components, I noticed that the `searchBar` of the `SearchViewController` of the (iDEAL) issuer list cannot be styled. See this screenshot of the sample app:
![issuer_list_searchbar](https://user-images.githubusercontent.com/1574034/223755318-8585cbe7-4969-485f-9247-79bb3c80c6f9.png)
I applied custom styles to the navigation bar as well as the list and its items. The search bar looks out of place.

**Describe the solution you'd like**
Two possible solutions:
- Add a configuration that can be used to hide the search bar of the issuer list. This does not solve the problem for users that need a search bar.
- Add a style for the search bar, for example as part of the `ListComponentStyle` assuming that lists tend to be searchable. The `IssuerListComponent` may then pass the search bar configuration to its `SearchViewController`.

**Describe alternatives you've considered**
- Use `UIAppearance` to style the search bar. This approach may work but is not an option in our current project.
- Traverse the iOS view hierarchy and hide or style the search bar. This approach may break in future updates of adyen-ios without notice. Not an option.

**Additional context**
None.

**Comment by erenbesel** [comment]
Hey @raginmari 
Thanks for the detailed explanation!

This should be fixed in https://github.com/Adyen/adyen-ios/pull/1162, which will be included in the next release!

**Comment by raginmari** [comment]
The change in the linked pull request addresses the background color / general tint color and makes the search bar look a lot better.

However, it does not make it possible to apply styles to the search bar font, text color, cursor color, and auxiliary views / icons. See the following screenshots (light and dark mode, with and without search text):

![issuer_list_search_empty_dark](https://user-images.githubusercontent.com/1574034/224001814-bbaa8818-8763-4520-baba-56266aa7a8c7.png)
![issuer_list_search_empty_light](https://user-images.githubusercontent.com/1574034/224001819-6fc2ebd4-5052-45c3-94d4-2ad72a4a5f7f.png)
![issuer_list_search_text_dark](https://user-images.githubusercontent.com/1574034/224001820-22808526-9d90-4af7-a488-e34a3fbd7bfd.png)
![issuer_list_search_text_light](https://user-images.githubusercontent.com/1574034/224001824-5b377edf-baab-4090-a1ef-e94d5a5108e5.png)

In light mode, the search bar is barely usable without custom styling.

**Comment by goergisn** [comment]
Closing ticket due to inactivity.

It looks like you were trying to force the component into dark mode.

You can always use:

```
let viewController = component.viewController
if #available(iOS 13.0, *) {
    viewController.overrideUserInterfaceStyle = .dark
}
presenter?.present(viewController: viewController, completion: nil)
```

Feel free to re-open if you have any other questions.
